### DISCUSSION TITLE
compile error

### DISCUSSION THREAD

**Discussion by tlzfighting** [contiki-ng/contiki-ng#2288] [Q&A]
When I TYPE this compile option “make TARGET=cc2538dk ” ，the following compile error occurs, please advise how to solve it

/usr/lib/gcc/arm-none-eabi/10.3.1/../../../arm-none-eabi/bin/ld:build/cc2538dk/obj/cc2538.ld:1529: warning: memory region `FLASH_FW=' not declared
/usr/lib/gcc/arm-none-eabi/10.3.1/../../../arm-none-eabi/bin/ld:build/cc2538dk/obj/cc2538.ld:1529: syntax error
collect2: error: ld returned 1 exit status
make: *** [../..//arch/cpu/arm/cortex-m/Makefile.customrules-cortex-m:11：build/cc2538dk/demo.elf]

**Comment by pjonsson** [comment]
In which directory?

**Reply by tlzfighting** [reply]
hello,     in this   directory:
/home/t/contiki-ng/examples/dev/dht11

**Reply by pjonsson** [reply]
That works for me on latest develop, inside the docker container:
```
user@b95cafbf177b:~/contiki-ng/examples/dev/dht11$ make TARGET=cc2538dk
  MKDIR     build/cc2538dk/obj/.deps
  ...
  CC        ../../../os/sys/stimer.c
  CC        ../../../os/sys/timer.c
  CC        dht11.c
  CC        ../../../arch/cpu/cc2538/cc2538.lds
  LD        build/cc2538dk/dht11.elf
  CP        build/cc2538dk/dht11.elf --> build/cc2538dk/dht11.cc2538dk
  CP        build/cc2538dk/dht11.cc2538dk --> dht11.cc2538dk
user@b95cafbf177b:~/contiki-ng/examples/dev/dht11$ 
```

I'm not sure if it will make any difference, but you can try a `make distclean` and see if that solves anything for you.

**Reply by tlzfighting** [reply]
hello,thanks for replying.   
I tried these commands "make distclean",  "make TARGET=cc2538dk",and the same problem occurred

**Reply by pjonsson** [reply]
Is the non-working build inside the docker image? If not, it would be great if you try to reproduce the problem inside docker so it can be fixed, instructions for running docker are here:

https://docs.contiki-ng.org/en/develop/doc/getting-started/Docker.html

**Reply by tlzfighting** [reply]
Hello,Sir.  Thanks for your reply.I have found the cause of the error. I did not use the command "contiker" before compiling.

*.elf file instead of *.bin file is generated when I use the compile command "make TARGET=cc2538dk"
 in the directory "contiki-ng/examples/hello-world". 
Is this a successful compilation result? If it is successful, how do I convert it to *.bin?  
Looking forward to your valuable reply!

![图片](https://user-images.githubusercontent.com/95647449/204106050-e833bade-07fc-4bc2-be69-b8b58fbc650b.png)


![图片](https://user-images.githubusercontent.com/95647449/204106165-08151fbe-6478-4b8e-a338-27dde6e8c63b.png)

**Reply by pjonsson** [reply]
The build system used to build bin files by default, but won't do that in v4.9, so you need to ask for the file explicitly. In this case, `make TARGET=cc2538dk build/cc2538dk/dht11.bin` will produce it for the dht11 example.

**Reply by tlzfighting** [reply]
I have got it ! Thanks for your valuable reply!

**Comment by alexrayne** [comment]
They are droped autogeneration of hex and bins for ARM cpus since v4.7. try
explicit "make aplication.hex" - maybe it still keep rules on that
generation

сб, 26 нояб. 2022 г., 22:41 tlzfighting ***@***.***>:

**Reply by tlzfighting** [reply]
Thanks for your valuable reply!
